Node.js/GettingStarted/nodejs_batch_client_sample.js

```diff
@@ -0,0 +1,188 @@
+                version:"latest"}
+var vmConfig = {imageReference:imgRef,
+    nodeAgentSKUId:"batch.node.ubuntu 14.04"}
+
+var numVms = 4;
+var vmSize = "STANDARD_A1";
+var clientPoolRequestID = uuid.v4();
+
+var poolConfig = {id:poolId, 
+    displayName:"Processing csv files",
+    vmSize:vmSize,
+    virtualMachineConfiguration:vmConfig,
+    targetDedicated:numVms,
+    enableAutoScale:false }
+var isPoolCreationError = false;
+var pool = batchClient.pool.add(poolConfig,function(error,result,request,response){
+    if(error !== null)
+    {
+        console.log(error.response);
+        isPoolCreationError = true;
+    }
+ });
+console.log("Submitted request to create pool...");
+
+var isPoolStateActive = false;
+var numRetries = 0;
+var jobCreated = false;
+
+while(isPoolStateActive == false && isPoolCreationError == false && numRetries < maxRetriesVal)
+{
+    var cloudPool = batchClient.pool.get(poolId,function(error,result,request,response)
+    {
+        if(error == null)
+        {
+            if(result.state == "active")
+            {
+                isPoolStateActive = true;
+                if(jobCreated==false)
+                {
+                    console.log("Pool created successfully, now submitting the job");
+                    // Create the Job 
+                    createJob(); 
+                    jobCreated = true;
+                }
+            }
+        }
+        else
+        {
+            if(error.statusCode==404)
+            {
+                /// Ignore 404 as the pool is still being created                                           
+            }
+            else
+            {
+                console.log("Error occured while pool creation : " + error.response);
+                isPoolCreationError = true;
+            }
+        }
+    });
+    numRetries++;
+    sleep(sleepInterval);
+}
+
+function createJob()
+{
+    console.log("Creating job...");    
+    var jobPrepTaskConfig = {id:"installprereq",
+                                commandLine:"sudo sh startup_prereq.sh > startup.log",
+                                resourceFiles:[{'blobSource':sh_url,'filePath':'startup_prereq.sh'}],
+                                waitForSuccess:true,
+                                runElevated:true}
+    // Setting up Batch pool configuration
+    var poolConfigJob = {poolId:poolId}    
+    
+    var jobConfig = {id:jobId,
+        displayName:"process csv files",
+        jobPreparationTask:jobPrepTaskConfig,
+        poolInfo:poolConfigJob}
+    
+     // Adding Azure Batch job to the pool
+     var job = batchClient.job.add(jobConfig,function(error,result){
+        if(error !== null)
+        {
+            console.log(error);
+        }
+    });
+
+    
+    var isJobCreated = false
+    var isJobCreatedError = false
+    var numRetriesJob = 0;
+    var taskSubmitted = false;
+    while(numRetriesJob < maxRetriesVal && isJobCreated == false && isJobCreatedError == false)
```

Comment:
It gives errors based on the state. Chances of the errors are very high, if the subsequent jobs/tasks etc are submitted before the state is active. It takes pretty long.